src/server/models/listItem/providers/Lawyers.ts

```diff
@@ -0,0 +1,181 @@
+// Lawyers
+import { LawyersFormWebhookData } from "server/components/formRunner";
+import {
+  CountryName,
+  LawyerListItemCreateInput,
+  LawyerListItemGetObject,
+  ServiceType,
+} from "server/models/types";
+import {
+  createAddressGeoLocation,
+  createCountry,
+  getPlaceGeoPoint,
+} from "./../geoHelpers";
+import { getListIdForCountryAndType } from "server/models/helpers";
+import { startCase, toLower, uniq } from "lodash";
+import { logger } from "server/services/logger";
+import { prisma } from "server/models/db/prisma-client";
+import { ListItemWithAddressCountry } from "./types";
+import { legalPracticeAreasList } from "server/services/metadata";
+import {
+  checkListItemExists,
+  fetchPublishedListItemQuery,
+} from "server/models/listItem/providers/helpers";
+
+export async function createObject(
+  lawyer: LawyersFormWebhookData
+): Promise<LawyerListItemCreateInput> {
+  try {
+    const {
+      addressCountry,
+      addressLine1,
+      addressLine2,
+      areasOfLaw,
+      city,
+      familyName,
+      firstAndMiddleNames,
+      postcode,
+      ...rest
+    } = lawyer;
+    const country = await createCountry(addressCountry);
+    const geoLocationId = await createAddressGeoLocation(lawyer);
+
+    const listId = await getListIdForCountryAndType(
+      lawyer.country as CountryName,
+      ServiceType.lawyers
+    );
+
+    return {
+      type: ServiceType.lawyers,
+      isApproved: false,
+      isPublished: false,
+      listId,
+      jsonData: {
+        ...rest,
+        areasOfLaw: uniq(areasOfLaw ?? []),
+        contactName: `${firstAndMiddleNames} ${familyName}`,
+      },
+      address: {
+        create: {
+          firstLine: addressLine1,
+          secondLine: addressLine2,
+          postCode: postcode,
+          city,
+          country: {
+            connect: {
+              id: country.id,
+            },
+          },
+          geoLocation: {
+            connect: {
+              id: typeof geoLocationId === "number" ? geoLocationId : undefined,
+            },
+          },
+        },
+      },
+    };
+  } catch (error) {
+    logger.error(error);
+    throw error;
+  }
+}
+
+export async function create(
+  webhookData: LawyersFormWebhookData
+): Promise<ListItemWithAddressCountry> {
+  const exists = await checkListItemExists({
+    organisationName: webhookData.organisationName,
+    countryName: webhookData.country,
+  });
+
+  if (exists) {
+    throw new Error("Lawyer record already exists");
+  }
+
+  try {
+    const data = await createObject(webhookData);
+
+    return await prisma.listItem.create({
+      data,
+      include: {
+        address: {
+          include: {
+            country: true,
+          },
+        },
+      },
+    });
+  } catch (error) {
+    logger.error(`createLawyerListItem Error: ${error.message}`);
+    throw error;
+  }
+}
+
+export async function findPublishedLawyersPerCountry(props: {
```

Comment:
similar point to above: consider refactoring to be a generic "find" function that can be used anywhere.  A specialised "findPublishedLawyersPerCountry" can still be used but it will simply set the published = true criteria and pass into the generic function.  This ensures we don't have to duplicate SQL in multiple areas if we want to include unpublished records in the resultset, e.g., dashboard list item management page.

Reply:
agreed but not my code - paging this issue so we don't forget: #381